examples/rust/tcp-ping-pong.rs

```diff
@@ -68,6 +68,94 @@ fn mksga(libos: &mut LibOS, size: usize, value: u8) -> demi_sgarray_t {
+        Ok(qt) => qt,
+        Err(e) => panic!("accept failed: {:?}", e.cause),
+    };
+    match libos.wait(qt, None) {
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_ACCEPT => unsafe { qr.qr_value.ares.qd.into() },
+        Err(e) => panic!("operation failed: {:?}", e.cause),
+        _ => panic!("unexpected result"),
+    }
+}
+
+//======================================================================================================================
+// connect_and_wait()
+//======================================================================================================================
+
+/// Connects to a remote socket and wait for the operation to complete.
+fn connect_and_wait(libos: &mut LibOS, sockqd: QDesc, remote: SocketAddrV4) {
+    let qt: QToken = match libos.connect(sockqd, remote) {
+        Ok(qt) => qt,
+        Err(e) => panic!("connect failed: {:?}", e.cause),
+    };
+    match libos.wait(qt, None) {
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CONNECT => println!("connected!"),
+        Err(e) => panic!("operation failed: {:?}", e),
+        _ => panic!("unexpected result"),
+    }
+}
+
+//======================================================================================================================
+// push_and_wait()
+//======================================================================================================================
+
+/// Pushes a scatter-gather array to a remote socket and waits for the operation to complete.
+fn push_and_wait(libos: &mut LibOS, qd: QDesc, sga: &demi_sgarray_t) {
+    // Push data.
+    let qt: QToken = match libos.push(qd, sga) {
+        Ok(qt) => qt,
+        Err(e) => panic!("push failed: {:?}", e.cause),
+    };
+    match libos.wait(qt, None) {
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_PUSH => (),
+        Err(e) => panic!("operation failed: {:?}", e.cause),
+        _ => panic!("unexpected result"),
+    };
+}
+
+//======================================================================================================================
+// pop_and_wait()
+//======================================================================================================================
+
+/// Pops a scatter-gather array from a socket and waits for the operation to complete.
+fn pop_and_wait(libos: &mut LibOS, qd: QDesc, recvbuf: &mut [u8]) {
+    let mut index: usize = 0;
+
+    // Pop data.
+    while index < recvbuf.len() {
+        let qt: QToken = match libos.pop(qd) {
+            Ok(qt) => qt,
+            Err(e) => panic!("pop failed: {:?}", e.cause),
+        };
+        let sga: demi_sgarray_t = match libos.wait(qt, None) {
+            Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_POP => unsafe { qr.qr_value.sga },
+            Err(e) => panic!("operation failed: {:?}", e.cause),
+            _ => panic!("unexpected result"),
+        };
+
+        // Sanity check received data.
```

Comment:
Is this the right comment?  This block appears to be copying the data out to the `recvbuf`, not sanity checking it.

Reply:
Oh, this is indeed not the right comment. Thanks for noting, I have fixed it.